internal/juju/applications.go

```diff
@@ -1114,7 +1114,13 @@ func (c applicationsClient) ReadApplication(input *ReadApplicationInput) (*ReadA
 	usedResources := make(map[string]string)
 	for _, iResources := range resources {
 		for _, resource := range iResources.Resources {
-			usedResources[resource.Name] = strconv.Itoa(resource.Revision)
+			// For custom resources we special case the revision number so that subsequent runs of `terraform apply`
+			// use the value in state rather than the revision number.
```

Comment:
Let's be more specific on the comment:

`Per juju convention, -1, indicates that an integer value has not been set. Uploaded resources currently have no revision number.`